app/src/main/java/org/sil/storyproducer/controller/BaseController.kt

```diff
@@ -79,11 +79,15 @@ open class BaseController(
     }
 
     private fun onStoriesUpdated() {
-        if (Workspace.registration.complete) {
-            view.showMain()
-        } else {
-            view.showRegistration()
+        if (!Workspace.registration.complete) {
+            // DKH - 05/12/2021
+            // Issue #573: SP will hang/crash when submitting registration
```

Comment:
do you mean?: SP will hang/crash if you call view.showRegistration() directly and user submits registration